Задание UI, BankingProject Test

oINCLUDEo opened the pull request:
Выполнены все задания вместе с дополнительными

oINCLUDEo committed:
open page, set value postcode

open page, click to button, check visibility, set static value to postcode input

oINCLUDEo committed:
update pom,move allure-results,steps in allure

now allure-results in target directory, steps tracked in allure, restructured pom.xml

oINCLUDEo committed:
added random generate Post Code(javafaker)

oINCLUDEo committed:
added First Name generator, logs, new test step

added Post Code to First Name convertor and new step for test

oINCLUDEo committed:
sort test, add test_suite.xml, parallel testing

oINCLUDEo committed:
new allure-annotations(epic, feature, story)

small revision test_suite.xml

oINCLUDEo committed:
Merge branch 'main' of https://github.com/oINCLUDEo/PracticumSDET into dev

oINCLUDEo committed:
new test deletion customer from customers table

oINCLUDEo committed:
separate BasePage, moved some functions to step

now BasePage removed and separate to BankManagerPage, AddCustomerPage, CustomersPage. Functions getClientForDeletion, getCustomersFirstNames, getCustomersLastNames was reworked to Steps

oINCLUDEo committed:
cleaned code

oINCLUDEo committed:
added Submitting Form,new step for input LastName

oINCLUDEo committed:
added headless mode, ci/cd with github actions

oINCLUDEo committed:
Added Github Pages Integration

oINCLUDEo committed:
better logs

oINCLUDEo committed:
Update README.md

(cherry picked from commit 5bdf3457e1135078f5667511a13e1106b1020d29)

oINCLUDEo committed:
histories for allure

bondarenkokate73 commented on src/test/java/pages/CustomersPage.java:
```diff
@@ -0,0 +1,160 @@
+    @Step("Получение списка имен Customers")
+    public List<String> getCustomersFirstNames() {
+        List<String> firstNames = customersTableRows.stream()
+                .map(row -> row.$("td:nth-child(1)").getText())
```
1. Не рекомендуется в методах напрямую обращаться к локаторам
2. Категорически не рекомендуется привязываться статически к порядковому номеру элемента, потому что оно в любой момент может поменяться
Если правильно найти локатор, то список имен найдется и без данных махинаций, останется только текст из них получить

oINCLUDEo committed:
revision from review, updated ci/cd

Deleted some empty lines, fixed static locators to elements, locators removed from methods, github pages now use actions not branch for page

oINCLUDEo commented:
Замечания учтены
